va/va.go

```diff
@@ -211,6 +233,8 @@ func (d *dialer) Dial(_, _ string) (net.Conn, error) {
 
 		// Otherwise, we note that we tried an address and fall back to trying IPv4
 		d.record.AddressesTried = append(d.record.AddressesTried, d.record.AddressUsed)
+		// Reconstruct the underlying dialer to get a new timeout for the second request
+		realDialer = d.realDialer()
```

Comment:
Oh wait, is this  necessary with the call on L257? It seems like you'll be constructing three dialers but only using the first and third?